### PR TITLE
Fix the issue #2322 to throw error for invalid epsilon argument passed.

### PR DESCRIPTION
**Pull request by mritunjaysharma394**
Fix the issue #2322 .
Implements producing an error for epsilon passed not in the range [0,1].

### CODE REVIEW

**Comment by mritunjaysharma394** [comment]
Please @jackgerrits review the PR of the issue #2322 and also do tell what could be the reason behind Windows CI Build failing?

**Comment by adityauser** [comment]
I don't think its the correct way to solve this issue.
You should check the epsilon conditions in these function: 
https://github.com/VowpalWabbit/vowpal_wabbit/blob/e1a05f483b0c9e35f6a039ed848433d24b6843c6/explore/explore_internal.h#L38
https://github.com/VowpalWabbit/vowpal_wabbit/blob/e1a05f483b0c9e35f6a039ed848433d24b6843c6/explore/explore_internal.h#L158
@jackgerrits what is your opinion?

**Comment by jackgerrits** [comment]
Enforcing epsilon in those functions may be a good thing. But I think this is something that should be caught early, and we can do so while validating command line options.

File: vowpalwabbit/cb_explore.cc [hunk 1]
```diff
@@ -302,6 +302,11 @@ base_learner* cb_explore_setup(options_i& options, vw& all)
     ss << data->cbcs.num_actions;
     options.insert("cb", ss.str());
   }
+  
+  if (data->epsilon < 0.0 || data->epsilon > 1.0)
+  {
+    THROW("cb_explore_epsilon must be in [0,1]");
```
Comment (jackgerrits) [inline vowpalwabbit/cb_explore.cc]: The use may not know what `cb_explore_epsilon` is, it doesn't correspond to an option they inputted. Can you make the error message clearer with that in mind?

**Comment by jackgerrits** [comment]
The windows failure is a flaky test, I'll restart

File: vowpalwabbit/cb_explore.cc (hunk 1 above)
Comment (mritunjaysharma394) [reply vowpalwabbit/cb_explore.cc]: Yes, I have modified the error messages. Please review them, thanks!

**Comment by mritunjaysharma394** [comment]
I think now it looks good to merge. What say @jackgerrits ?

**Pull request merged**